server/plugin/flows.go

```diff
@@ -694,13 +694,16 @@ func (fm *FlowManager) submitWebhook(f *flow.Flow, submitted map[string]interfac
 		return "", nil, nil, errors.Wrap(err, "failed to create hook")
 	}
 
-	select {
-	case event := <-ch:
-		if *event.HookID == *hook.ID {
-			break
+	var found bool
+	for !found {
+		select {
+		case event, ok := <-ch:
+			if ok && event != nil && *event.HookID == *hook.ID {
+				found = true
```

Comment:
I know the loop is set to use `found` as its terminating value, but do we want to include the `break` here for readability, or does that cause us to exit the `switch` only?

Reply:
`break` only exists the `select` statement. 

> A "break" statement terminates execution of the innermost ["for"](https://go.dev/ref/spec#For_statements), ["switch"](https://go.dev/ref/spec#Switch_statements), or ["select"](https://go.dev/ref/spec#Select_statements) statement within the same function.

https://go.dev/ref/spec#Break_statements